filebeat/input/file/glob.go

```diff
@@ -0,0 +1,60 @@
+package file
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// globPattern detects the use of "**" and expands it to standard glob patterns up to a max depth
+func globPatterns(pattern string, doubleStarPatternDepth uint8) ([]string, error) {
+	if doubleStarPatternDepth == 0 {
+		return []string{pattern}, nil
+	}
+	var patterns []string
+	isAbs := filepath.IsAbs(pattern)
+	patternList := strings.Split(pattern, string(os.PathSeparator))
```

Comment:
Ah interesting, didn't know that. So the windows case is basically a superset, let me handle that too.